tests/test_mail.py

```diff
@@ -0,0 +1,67 @@
+from aea.mail.base import Envelope, MailBox, InBox, OutBox
+
+from aea.protocols.base.message import Message
+from aea.protocols.oef.serialization import DEFAULT_OEF, OEFSerializer
+from aea.channels.oef.connection import OEFConnection
+from aea.channels.local.connection import LocalNode, OEFLocalConnection
+from aea.protocols.base.serialization import ProtobufSerializer, JSONSerializer
+from queue import Queue
+import pytest
+
+
+def test_envelope_initialisaztion() : 
+	msg = Message(content='hello')
+	message_bytes = ProtobufSerializer().encode(msg)
+	envelope = Envelope(to="Agent1", sender="Agent0", protocol_id="my_own_protocol", message=message_bytes)
+	
+	envelope.to = "Agent1"
+	envelope.sender = "Agent0"
+	envelope.protocol_id = "my_own_protocol"
+	envelope.message = b"hello"
+	
+	assert (envelope,"Cannot generate a new envelope")
+
+def test_envelope_empty_receiver() : 
+	to_adr = []
+	msg = Message(content="hello")
+	message_bytes = ProtobufSerializer().encode(msg)
+	envelope = Envelope(to=to_adr, sender="Agent0", protocol_id="my_own_protocol", message=message_bytes)
+	assert (envelope, "Passing an empty receiver. Testing try/except by passing a list as sender")
+
+def test_inbox_empty():
+	my_queue = Queue() 
+	_inbox = InBox(my_queue)
+	assert(_inbox.empty(), "Must return an empty inbox")
```

Comment:
Comment should be relevant in the fail case, i.e. "Does not return an empty inbox."